src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -20,10 +20,13 @@ class GFAL2Impl(StageOutImpl):
 
     def __init__(self, stagein=False):
         StageOutImpl.__init__(self, stagein)
-        self.setups = "env -i X509_USER_PROXY=$X509_USER_PROXY; "
-        self.setups += 'date "+%Y-%m-%dT%H:%M:%S";'
-        self.removeCommand = "%s gfal-rm -vvv -t 600 " % self.setups
-        self.copyCommand = "%s gfal-copy -vvv -t 2400 -T 2400 -p " % self.setups
+        # If we want to execute commands in clean shell, we can`t separate them with ';'.
+        # Next commands after separation are executed without env -i and this leads us with 
+        # mixed environment with COMP and system python.
+        # GFAL2 is not build under COMP environment and it had failures with mixed environment.
+        self.setups = "env -i X509_USER_PROXY=$X509_USER_PROXY sh -c '%s'"
+        self.removeCommand = self.setups % 'printenv; date; gfal-rm -vvv -t 600 %s '
```

Comment:
It's just a taste of code, I think ;) I can add it, but why adding a date everywhere in each plugin, while you can add it before executing the command, which is already done. Why you need this in multiple places?
https://github.com/dmwm/WMCore/blob/0b3086a3b359bdd2d1a75461c7f7f3f8a14ea4bf/src/python/WMCore/Storage/StageOutImpl.py#L175